src/libz.jl

```diff
@@ -23,6 +23,18 @@ mutable struct ZStream
     reserved::Culong
 end
 
+@assert typemax(Csize_t) ≥ typemax(Cuint)
+
+function zalloc(::Ptr{Cvoid}, items::Cuint, size::Cuint)::Ptr{Cvoid}
+    s, f = Base.Checked.mul_with_overflow(items, size)
+    if f
+        C_NULL
+    else
+        ccall(:jl_malloc, Ptr{Cvoid}, (Csize_t,), s%Csize_t)
+    end
+end
+zfree(::Ptr{Cvoid}, p::Ptr{Cvoid}) = ccall(:jl_free, Cvoid, (Ptr{Cvoid},), p)
```

Comment:
Since `zfree` is just a pass through to `jl_free` perhaps you should just provide the function pointer directly:

`cglobal(:jl_free)`.

Reply:
They have different signatures. `zfree` takes a pointer to an allocator context and a pointer to free, while `jl_free` just takes a pointer to free.